tests/test_cli.py

```diff
@@ -8,19 +7,21 @@
 from dsc.db.models import ItemSubmissionDB, ItemSubmissionStatus
 
 
-@patch("dsc.utilities.aws.s3.S3Client.files_iter")
```

Comment:
These `@patch` calls seemed to be hiding that `metadata.csv` would be returned unless explicitly excluded in `Workflow.exclude_prefixes` which was revealed by switching  to `moto`. But please correct me if I'm missing something!

Reply:
I was trying to understand how/why this didn't cause an issue when we were running those deposits, @ehanson8 -- in other words, why has it been okay that `f"{self.batch_path}metadata.csv"` was _not_ a part of `Workflow.exclude_prefixes`. This was because:

> We were _previously_ only using `S3Client.files_iter` to get the bitstream URIs for a given `item_identifier`; this required passing `item_identifier` into the `S3Client.files_iter`, which effectively filtered out the metadata CSV file. 

However, with the refactor, we are now using `S3Client.files_iter` in `Workflow.get_batch_bitstream_s3_uris`, which would capture the metadata CSV file unless explicitly set in `Workflow.exclude_prefixes`.

TLDR: Excellent catch! :)